src/plugin/managerbase.js

```diff
@@ -0,0 +1,471 @@
+                pluginContext = {
+                    branchName: context.branchName,
+                    commitHash: context.commitHash,
+
+                    rootNode: null,
+                    activeNode: null,
+                    activeSelection: null,
+                    META: {},
+                    namespace: context.namespace || '',
+
+                    project: context.project,
+                    projectId: context.project.projectId,
+                    projectName: storageUtil.getProjectNameFromProjectId(context.project.projectId),
+                    core: new Core(context.project, {
+                        globConf: gmeConfig,
+                        logger: self.logger.fork('core')
+                    })
+                };
+
+            self.logger.debug('loading context');
+            getBranchHash(pluginContext.project, pluginContext.branchName)
+                .then(function (branchHash) {
+                    pluginContext.commitHash = context.commitHash || branchHash;
+                    if (!pluginContext.commitHash) {
+                        throw new Error('Neither commitHash nor branchHash from branch was obtained, branchName: [' +
+                            context.branchName + ']');
+                    }
+                    return Q.ninvoke(pluginContext.project, 'loadObject', pluginContext.commitHash);
+                })
+                .then(function (commitObject) {
+                    var rootDeferred = Q.defer();
+                    self.logger.debug('commitObject loaded', {metadata: commitObject});
+                    pluginContext.core.loadRoot(commitObject.root, function (err, rootNode) {
+                        if (err) {
+                            rootDeferred.reject(err);
+                        } else {
+                            self.logger.debug('rootNode loaded');
+                            rootDeferred.resolve(rootNode);
+                        }
+                    });
+
+                    return rootDeferred.promise;
+                })
+                .then(function (rootNode) {
+                    pluginContext.rootNode = rootNode;
+                    // Load active node
+                    return self.loadNodeByPath(pluginContext, context.activeNode || '');
+                })
+                .then(function (activeNode) {
+                    pluginContext.activeNode = activeNode;
+                    self.logger.debug('activeNode loaded');
+                    // Load active selection
+                    return self.loadNodesByPath(pluginContext, context.activeSelection || []);
+                })
+                .then(function (activeSelection) {
+                    pluginContext.activeSelection = activeSelection;
+                    self.logger.debug('activeSelection loaded');
+                    // Load meta nodes
+                    var metaIds = pluginContext.core.getMemberPaths(pluginContext.rootNode, 'MetaAspectSet');
```

Comment:
this should be replaced with the getAllMetaNodes synchronous call (probably faster...)